[BD-38]  feat: add flag to toggle learners tab in the discussions MFE [TNL-9481][BB-5421]

Cup0fCoffee opened the pull request:
## Description

Add a flag to toggle learners tab in the Discussions MFE.

## Supporting information

[TNL-9481](https://openedx.atlassian.net/browse/TNL-9481)
[BB-5421](https://tasks.opencraft.com/browse/BB-5421)

## Testing instructions

* Check that `<LMS URL>/api/discussion/v1/courses/<course-id>` returns `False` for `learners_tab_enabled` by default
* Go to `<LMS URL>/admin/waffle/flag/` and add a flag with `discussions.enable_learners_tab_in_discussions_mfe` name
* Check that `<LMS URL>/api/discussion/v1/courses/<course-id>` now returns `True` for `learners_tab_enabled`

## Deadline

"None"

Cup0fCoffee committed:
feat: add flag to toggle learners tab

openedx-webhooks commented:
Thanks for the pull request, @Cup0fCoffee! I've created [BLENDED-1126](https://openedx.atlassian.net/browse/BLENDED-1126) to keep track of it in Jira.   More details are on the [BD-38](https://openedx.atlassian.net/wiki/spaces/COMM/pages/2790490201/BD-38+Discussions+In-Context+Discussion+Moderation+Phase+2) project page.


When this pull request is ready, tag your edX technical lead.


<!-- comment:welcome-blended -->
<!-- jenkins ok to test -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

Cup0fCoffee commented on lms/djangoapps/discussion/toggles.py:
```diff
@@ -13,3 +13,12 @@
+# .. toggle_description: Waffle flag to enable learners tab in the new MFE experience for discussions
+# .. toggle_use_cases: temporary, open_edx
+# .. toggle_creation_date: 2022-02-21
+# .. toggle_target_removal_date: ?
```
@xitij2000 I assume that this is going to be a temporary flag. What should be the `toggle_target_removal_date`?

xitij2000 replied:
@Cup0fCoffee we can keep it for 3 months and then revisit if needed.

xitij2000 commented on openedx/core/djangoapps/discussions/serializers.py:
```diff
@@ -406,7 +407,8 @@ def to_representation(self, instance: CourseDiscussionSettings) -> dict:
             'division_scheme': instance.division_scheme,
-            'available_division_schemes': available_division_schemes(course_key)
+            'available_division_schemes': available_division_schemes(course_key),
+            'learners_tab_enabled': ENABLE_LEARNERS_TAB_IN_DISCUSSIONS_MFE.is_enabled(course_key),
```
@Cup0fCoffee I don't think this is the right place for this since we want this flag to be read by the MFE, and all users. So it shold be in `lms/djangoapps/discussion/rest_api/views.py` as part of the discussion CourseView endpoint.

xitij2000 reviewed (changes requested):
I think this needs to be moved to another place, but other than that it seems fine.

xitij2000 reviewed (approved):
:+1:  Looks good! 

- [x] I tested this
- [x] I read through the code
- [x] Includes documentation

asadazam93 commented:
@Cup0fCoffee Can you please fix the conflicts?

natabene commented:
@Cup0fCoffee Thank you for your contribution. Please let me know once it is ready for our review.

Cup0fCoffee commented:
@asadazam93 I've rebased the branch and fixed the conflicts. There is a [check failing now](https://github.com/openedx/edx-platform/runs/5397732570?check_suite_focus=true):
```
ERROR: Could not find a version that satisfies the requirement pact-python==1.5.0
ERROR: No matching distribution found for pact-python==1.5.0
```
I don't think it's related to my changes. Can we rerun the check or should I try rebasing and pushing again?

@natabene Unless the failing check I mentioned above is an issue, the PR is ready for review.

asadazam93 commented:
@Cup0fCoffee Can you please fix the tests?

xitij2000 commented:
@Cup0fCoffee  Please add the [BD-38] tag here. Also you can rerun the tests by force pushing the branch. Just rebase on the latest changes and push again, the tests will run again.

Cup0fCoffee commented:
@asadazam93 Done.

openedx-webhooks commented:
@Cup0fCoffee 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/edx-platform&entry.867055334=https://github.com/openedx/edx-platform/pull/29955&entry.1484655318=https://github.com/Cup0fCoffee&entry.752974735=2022-02-21+02:53&entry.1917517419=2022-03-07+06:14&entry.2133058324=Yes) so we can improve your experience in the future.

<!-- comment:end_survey -->
